lms/djangoapps/course_structure_api/v0/views.py

```diff
@@ -0,0 +1,236 @@
+""" API implementation for course-oriented interactions. """
+
+import logging
+
+from django.conf import settings
+from django.http import Http404
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.generics import RetrieveAPIView, ListAPIView
+from xmodule.modulestore.django import modulestore
+from opaque_keys.edx.keys import CourseKey
+
+from courseware.access import has_access
+from openedx.core.djangoapps.content.course_structures import models
+from openedx.core.lib.api.permissions import IsAuthenticatedOrDebug
+from openedx.core.lib.api.serializers import PaginationSerializer
+from courseware import courses
+from course_structure_api.v0 import serializers
+from student.roles import CourseInstructorRole, CourseStaffRole
+
+
+log = logging.getLogger(__name__)
+
+
+class AuthMixin(object):
```

Comment:
This is only used in one place, so this functionality could be moved into CourseViewMixin